### PR TITLE
RFE: list noaudit exemptions

### PR DESCRIPTION
**Pull request by cgzones**
In the general case a rejected capability check will result in an audit event.  There are however some instances in the kernel where denied capability checks are not audited, which could lead to differences in behavior between enforcing and permissive mode.

Document this fact and list (hopefully) all occurrences in kernel v6.4.

Inspired by https://patchwork.kernel.org/project/selinux/patch/20230718115607.65652-1-omosnace@redhat.com/
/cc @WOnder93

### CODE REVIEW

File: src/auditing.md
Comment (dburgener) [inline src/auditing.md]: spelling: "completeness"

File: src/auditing.md
Comment (dburgener) [inline src/auditing.md]: Wouldn't it make more sense to enumerate the callers of `ptracer_capable()` in this list?  This seems like more of a utility function than the sort of call-site to enumerate here.  Grepping ptracer_capable looks like it only has 3 callers, so it wouldn't bloat the list too much to enumerate them instead of listing this call site.

File: src/auditing.md
Comment (dburgener) [inline src/auditing.md]: s/execpt/except if/

File: src/auditing.md
Comment (dburgener) [inline src/auditing.md]: I haven't fully grokked `do_task_stat()` at all, but is this really the only difference?  I see the comment about that, but then `permitted` is used quite a few more times in the function, and I don't know the /proc portion of the kernel well enough to understand what's going on in the rest of it without a lot more digging.

**Comment by jbrindle** [comment]
@cgzones Any chance you can publish an update to this based on Daniel's feedbacj above, or would you prefer us to pick it up?

File: src/auditing.md
Comment (cgzones) [reply src/auditing.md]: Listed the conditions

File: src/auditing.md
Comment (cgzones) [reply src/auditing.md]: Listed all affected fields

**Review by jbrindle — APPROVED** [review]

**Comment by jbrindle** [comment]
reviewed-by: Joshua Brindle <brindle@gmail.com>

**Comment by dburgener** [comment]
@jbrindle In the future, would you be able to wait longer before a merge?  I would think that typically someone who has reviewed an earlier version may want to review a later version as well, but with just 3 hours after the update, I hadn't gotten to it yet.  (It's also the workflow described in https://github.com/SELinuxProject/selinux-notebook/blob/main/MAINTAINER_PROCESS.md to wait for two maintainer acks unless "a reasonable period of time (for example a delay of over two weeks)" has passed)

**Pull request merged**
